src/internal_rules.jl

```diff
@@ -558,3 +558,418 @@ function EnzymeRules.reverse(
+            nothing
+        else
+            error("Error in forward ldiv! Enzyme rule $(typeof(fact)) $(typeof(B)).")
+        end
+        retdval
+    end
+    if RT <: Const
+        return retval
+    elseif (RT <: DuplicatedNoNeed) || (RT <: BatchDuplicatedNoNeed)
+        return retdval
+    else
+        return BatchDuplicated(retval, ntuple(i -> bretdval[i], Val(N)))
+    end
+end
+
+function EnzymeRules.augmented_primal(
+    config,
+    func::Const{typeof(cholesky)},
+    RT::Type{<:Union{Const, Duplicated, BatchDuplicatedNoNeed, BatchDuplicated}},
+    A::Union{Const, Duplicated, BatchDuplicated};
+    kwargs...
+)
+    fact = if EnzymeRules.needs_primal(config)
+        cholesky(A.val; kwargs...)
+    else
+        nothing
+    end
+    # dfact would be a dense matrix, prepare buffer
+    dfact = if EnzymeRules.width(config) == 1
+        Cholesky(Matrix(fact), 'L', 0)
+    else
+        ntuple(Val(EnzymeRules.width(config))) do i
+            Base.@_inline_meta
+            Cholesky(Matrix(fact), 'L', 0)
+        end
+    end
+    return EnzymeRules.AugmentedReturn(fact, dfact, (dfact,))
+end
+
+function EnzymeRules.reverse(
+    config,
+    ::Const{typeof(cholesky)},
+    dret,
+    cache,
+    A;
+    kwargs...
+)
+    (_dfact,) = cache
+    for b in 1:EnzymeRules.width(config)
+        dA = if EnzymeRules.width(config) == 1
+            A.dval
+        else
+            A.dval[b]
+        end
+        dfact = if EnzymeRules.width(config) == 1
+            _dfact
+        else
+            _dfact[b]
+        end
+        dA .+= dfact.factors
+    end
+    return (nothing,)
+end
+
+function EnzymeRules.augmented_primal(
+        config,
+        func::Const{typeof(\)},
+        RT::Type{<:Union{Const, DuplicatedNoNeed, Duplicated, BatchDuplicatedNoNeed, BatchDuplicated}},
+
+        fact::Annotation{C},
+        B::Union{Const, DuplicatedNoNeed, Duplicated, BatchDuplicatedNoNeed, BatchDuplicated};
+        kwargs...
+) where {C <: Cholesky}
+    x = copy(B.val)
+    primal = if EnzymeRules.needs_primal(config)
+        x
+    else
+        nothing
+    end
+    ldiv!(fact.val, x)
+    nobatched = EnzymeRules.width(config) == 1 ? true : false
+    shadow = nobatched ? zeros(size(B.val)) :
+        ntuple(Val(EnzymeRules.width(config))) do i
+            Base.@_inline_meta
+            zeros(size(B.val))
+        end
+    buffer = nobatched ? zeros(size(B.val)) :
+        ntuple(Val(EnzymeRules.width(config))) do i
+            Base.@_inline_meta
+            zeros(size(B.val))
+        end
+    return EnzymeRules.AugmentedReturn(primal, shadow, (x, shadow, buffer))
+end
+
+function EnzymeRules.reverse(
+    config,
+    func::Union{Const{typeof(\)}},
+    dret,
+    cache,
+    _fact::Annotation{C},
+    _B::Annotation;
+    kwargs...
+) where {C <: Cholesky}
+
+    (x, _dx, _buffer) = cache
+    for b in 1:EnzymeRules.width(config)
+        dx = if EnzymeRules.width(config) == 1
```

Comment:
Same comment as above re zip for clarity.

Reply:
Fixed.